src/main/java/duke/TaskStorage.java

```diff
@@ -0,0 +1,38 @@
+package duke;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskStorage {
+    private List<Task> taskstorage;
+    // Tasks are indexed from 0 in taskstorage.
+    private int ind = 0; //Number of Current Tasks
+
+    public TaskStorage() {
+        taskstorage = new ArrayList<Task>();
+    }
+    public void addTask(Task t) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + t);
+        this.taskstorage.add(t);
+        ind++;
+        System.out.println("Now you have " + ind + " task(s) in the list.");
```

Comment:
Perhaps you can move System.out.println(..) into a function in your Ui class to utilise it more, since the purpose of the Ui class is to handle all interactions with the user

Reply:
Noted with thanks!